pkg/apis/provisioning/v1alpha5/provisioner_validation.go

```diff
@@ -29,6 +29,7 @@ import (
 
 var (
 	SupportedNodeSelectorOps = []string{string(v1.NodeSelectorOpIn), string(v1.NodeSelectorOpNotIn)}
+	SupportedCapacityType    = []string{CapacityTypesOnDemand, CapacityTypesSpot}
```

Comment:
Whatever our validation strategy is, it should be applied equally to all well known labels (zone, arch, os, capacity type, instance type). The current validation logic happens at runtime as part of requirements math: https://github.com/aws/karpenter/blob/main/pkg/controllers/provisioning/controller.go#L141

I'm not sure we want to do api time validation, as the zones, etc may change over time given subnet configuration, etc. Instead, runtime validation (requirements math) is working pretty well. 

The key issue from the referenced bug is that 
```
2022-01-13T21:43:19.333Z	ERROR	controller.provisioning	Failed to find instance type option(s) for [default/inflate-6b88c9fb68-8vvs5 default/inflate-6b88c9fb68-22hlt default/inflate-6b88c9fb68-dfzgm default/inflate-6b88c9fb68-ttrhh]	{"commit": "5047f3c", "provisioner": "default"}
```
is an unintuitive error message. I'd focus on making sure that we error with something related to capacity type (or zone, etc if relevant), rather than trying to do api time valiation..

WDYT?